micro: copy op DEPTH_TO_SPACE kernel from lite

rkuester opened the pull request:
Copy the kernel and test for operator DEPTH_TO_SPACE from
tensorflow/lite/kernels at 49524d6 without making modifications.
Adapt to micro and add to the build later.

This PR is part of the work to port operator DEPTH_TO_SPACE
from lite to micro, as tracked in issue #46025.

rkuester committed:
micro: copy op DEPTH_TO_SPACE kernel from lite

Copy the kernel and test for operator DEPTH_TO_SPACE from
tensorflow/lite/kernels at 49524d68 without making modifications.
Adapt to micro and add to the build later.

rkuester committed:
micro: clang-format copy of DEPTH_TO_SPACE

clang-format the copy of DEPTH_TO_SPACE added in the previous
commit so that it passes CI checks, even though the files are not
yet included in the build.

rkuester committed:
micro: fixup use of ReportError in DEPTH_TO_SPACE

Change use of ReportError() to TF_LITE_KERNEL_LOG to satisfy an
internal CI check, even though this code is not yet included in
the build.

advaitjain commented on tensorflow/lite/micro/kernels/depth_to_space_test.cc:
```diff
@@ -0,0 +1,110 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
```
we're hitting an annoying discrepancy between internal and extenal clang-formatting.

The solution here is to delete the gmock and gtest includes.

The underlying problem is that when the code is imported internally, these includes become:

```cc
#include "/internal/path/to/gmock/gmock.h"
```

and then clang-format wants them to be in the lower block (i.e. next to `"flatbuffers/flatbuffers.h"`.

So the internal and external ordering is conflicting.

We only recently added clang-format checks externally which might explain why we haven't seen this before.

rkuester replied:
Okay, attempting in c30b352.

advaitjain replied:
https://github.com/tensorflow/tensorflow/pull/46298 will help catch this via the external CI.

rkuester committed:
micro: delete gmock/gtest includes in copy of DEPTH_TO_SPACE

Delete the gmock and gtest includes in the currently inactive
copy of op DEPTH_TO_SPACE. There is an annoying difference
between internal and external versions of the code that causes
clang-formatting errors in CI checks. When the code is imported
internally, the gmock and gtest includes become:

    #include "/internal/path/to/gmock/gmock.h"

and then clang-format wants them to be in the lower include block
(i.e., next to "flatbuffers/flatbuffers.h").

The solution for now is to delete gmock and gtest includes when
making these operator copies.
